test/Microsoft.Extensions.Localization.Tests/ResourceManagerStringLocalizerFactoryTest.cs

```diff
@@ -111,6 +122,34 @@ public void Create_FromType_ReturnsNewResultForDifferentType()
             Assert.NotSame(result1, result2);
         }
 
+        [Fact]
+        public void Create_ResourceLocationAttribute_UsesRootNamespace()
```

Comment:
Need a test enshrining the has `RootNamespaceAttribute` / doesn't have `ResourceLocationAttribute` behaviour.

Reply:
I don't know what you mean by this, could you expand?

Reply:
I mean this test illustrates what happens when both attributes are in play. From your comments on the product code, I understand that `[assembly: RootNamespace(...)]` is a no-op when there's no `ResourceLocationAttribute` in play. Need a test demonstrating that.